pkg/controller/apiserver/handlers/handler_test.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"encoding/json"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware-tanzu/antrea/pkg/controller/apiserver/handlers/endpoint"
+	antreatypes "github.com/vmware-tanzu/antrea/pkg/controller/types"
+	queriermock "github.com/vmware-tanzu/antrea/pkg/querier/testing"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type TestCase struct {
+	// query arguments sent to handler function
+	handlerArgs     []string
+	expectedStatus  int
+	// expected result written by handler function
+	expectedContent endpoint.Policies
+	// arguments of call to mock
+	argsMock       []string
+	// results of call to mock
+	appliedMock    []antreatypes.NetworkPolicy
+	egressMock     []antreatypes.NetworkPolicy
+	ingressMock    []antreatypes.NetworkPolicy
+}
+
+var responses = []endpoint.Policies{
+	{
+		Applied: []antreatypes.NetworkPolicy{},
+		Egress: []antreatypes.NetworkPolicy{},
+		Ingress: []antreatypes.NetworkPolicy{},
+	},
+	{
+		Applied: []antreatypes.NetworkPolicy{
+			{
+				SpanMeta:  antreatypes.SpanMeta{},
+				Name:      "policy1",
+			},
+		},
+		Egress: []antreatypes.NetworkPolicy{},
+		Ingress: []antreatypes.NetworkPolicy{},
+	},
+	{
+		Applied: []antreatypes.NetworkPolicy{
+			{
+				SpanMeta:  antreatypes.SpanMeta{},
+				Name:      "policy1",
+			},
+			{
+				SpanMeta:  antreatypes.SpanMeta{},
+				Name:      "policy2",
+			},
+		},
+	},
+}
+
+func TestIncompleteOrInvalidArguments(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	// sample selector arguments (right now, only supports podname and namespace)
+	podName, namespace := "podName", "namespace"
+	// outline test cases with expected behavior
+	testCases := map[string]TestCase{
+		"Responds with empty list given no name and no namespace": {
+			handlerArgs:           []string{"", ""},
+			expectedStatus:  http.StatusOK,
```

Comment:
is `StatusOK` the right thing to do here? Would you say these are error cases or no?

Reply:
Maybe http.StatusNoContent would be correct here?

Reply:
I believe we have been using `http.StatusBadRequest` consistently for such cases

Reply:
I have made the changes